### PR TITLE
Remove documentation for build retention

### PR DESCRIPTION
**Pull request by jordandcarter**
Build retention will no longer be configurable and so this PR removes the documentation for it.

Build export used to live on the build retention page so I've moved it to it's own page build-export and updated the nav.

### CODE REVIEW

File: data/nav.yml [hunk 1]
```diff
@@ -224,8 +224,8 @@
           path: "pipelines/artifacts"
         - name: "Pipeline tags"
           path: "pipelines/tags"
-        - name: "Build retention"
-          path: "pipelines/build-retention"
+        - name: "Build export"
```
Comment (mbelton-buildkite) [inline data/nav.yml:227]: This is a feature name, which assumes people know what it is. I think we could use a title that uses these key words but speaks more to a user's goal:

```suggestion
        - name: "Export build data"
```

What do you think?

File: pages/pipelines/build_export.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Configuring build export
```
Comment (mbelton-buildkite) [inline pages/pipelines/build_export.md:1]: If you update the nav, also update this heading.

File: pages/pipelines/build_export.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+Each [Buildkite plan](https://buildkite.com/pricing) has a maximum build retention
+period, after which your oldest builds will be automatically deleted.
```
Comment (mbelton-buildkite) [inline pages/pipelines/build_export.md:4]: tightening slightly while we're here:

```suggestion
period, after which your oldest builds are automatically deleted.
```

File: data/nav.yml [hunk 4]
```diff
@@ -224,8 +224,8 @@
           path: "pipelines/artifacts"
         - name: "Pipeline tags"
           path: "pipelines/tags"
-        - name: "Build retention"
-          path: "pipelines/build-retention"
```
Comment (mbelton-buildkite) [inline data/nav.yml:228]: Can you add a redirect for this URL to the new one? I believe they're defined in [routes.rb](https://github.com/buildkite/docs/blob/main/config/routes.rb) (Is that right, @dannymidnight?)

File: data/nav.yml (hunk 1 above)
Comment (jordandcarter) [reply data/nav.yml:227]: Yeah that makes sense 👍

File: data/nav.yml (hunk 4 above)
Comment (jordandcarter) [reply data/nav.yml:228]: Yeah that looks right, I've added in the redirect

**Review by dannymidnight — APPROVED** [review]
LGTM

**Review by mbelton-buildkite — APPROVED** [review]
Nice!

**Pull request merged**